recipes/icc/all/conanfile.py

```diff
@@ -0,0 +1,124 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration, ConanException
+import os
+
+
+class ICCConan(ConanFile):
+    name = 'icc'
+    homepage = 'https://github.com/redradist/Inter-Component-Communication'
+    license = 'MIT'
+    url = 'https://github.com/conan-io/conan-center-index'
+    description = "I.C.C. - Inter Component Communication, This is a library created to simplify communication between " \
+                  "components inside of single application. It is thread safe and could be used for creating " \
+                  "components that works in different threads. "
+    topics = ("thread-safe", "active-object", "communication")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        'fPIC': [True, False],
+    }
+    default_options = {
+        'shared': False,
+        'fPIC': True,
+    }
+    generators = "cmake", "cmake_find_package", "cmake_find_package_multi"
+    exports_sources = ['CMakeLists.txt', 'patches/*']
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 11
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "15",
+            "apple-clang": "9.4",
+            "clang": "3.3",
+            "gcc": "4.9.4"
+        }
+
+    def _is_runtime_and_shared_option_compatible(self):
+        return (self.settings.compiler.runtime == "MT" and not self.options.shared) or \
```

Comment:
Typically static lib and MD are used together, this currently prevents the default profile from working 🤔

Reply:
By some reason I had an issue with compiling package as dynamic library (ICC.dll), but test_package expected it to be static library (ICC.lib) and test_package failed

Reply:
Did you export symbols? https://docs.microsoft.com/en-us/cpp/cpp/dllexport-dllimport?view=msvc-160

Reply:
Yeah, that was an issue :(